vmdb/app/services/orchestration_template_dialog_service.rb

```diff
@@ -0,0 +1,151 @@
+class OrchestrationTemplateDialogService
+  def create_dialog(dialog_label, template)
+    ActiveRecord::Base.transaction do
+      dialog = Dialog.create(:label => dialog_label, :buttons => "submit,cancel")
+      tab = DialogTab.create(:display => "edit", :label => "Basic Information", :dialog => dialog, :position => 0)
```

Comment:
We should avoid using create on the dialog models and instead build out the dialog in memory first and then use ```.save!``` to commit it to the db.  That way if the dialog or associated models fail validation the dialog will get rolled back and there will not be a partially created dialog in the db.  Note: Using ```.save!``` will raise the error message where "save" will silently fail.

These two lines would change to become:
```ruby
dialog = Dialog.new(:label => dialog_label, :buttons => "submit,cancel")
tab    = dialog.dialog_tabs.build(:display => "edit", :label => "Basic Information", :position => 0)
```

Also note that you no longer have to pass the :dailog => dialog property to the build method.